pkgs/top-level/all-packages.nix

```diff
@@ -19568,6 +19568,10 @@ with pkgs;
 
   nethack = callPackage ../games/nethack { };
 
+  nethack-x11 = nethack.override {
```

Comment:
Is it okay to leave this out? I much prefer to have users set up these kind of overrides in their configs. You can definitely add documentation to the Nixpkgs manual if you think it needs to be more disoverable.

Reply:
I don't know how correct my approach is, but I used the crawlTiles and crawl packages as a sample. Everything is done exactly the same way there.

What would you suggest to do in this case? Basically, I could make a separate nix expression for the x11 version and maintain it separately.

Reply:
No -  I definitely think it's good to reuse the nethack expression - I just don't want it to go in all-packages.nix.

Reply:
Could you give an example of packages where everything is done the right way?

Reply:
@matthewbauer 
```
nethack-x11 = callPackage ../games/nethack { x11Mode = true; };
```
Perhaps it would be better?

Reply:
I've made some changes.